apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -198,7 +202,46 @@ namespace MWRender
         mSceneRoot = sceneRoot;
         sceneRoot->setStartLight(1);
 
-        mRootNode->addChild(sceneRoot);
+        osg::ref_ptr<osgShadow::ShadowedScene> shadowedScene (new osgShadow::ShadowedScene);
+
+        osgShadow::ShadowSettings* settings = shadowedScene->getShadowSettings();
+        settings->setLightNum(0);
+        settings->setCastsShadowTraversalMask(Mask_Scene|Mask_Actor|Mask_Player|Mask_Terrain);
+        settings->setReceivesShadowTraversalMask(~0u);
+
+        //settings->setShadowMapProjectionHint(osgShadow::ShadowSettings::PERSPECTIVE_SHADOW_MAP);
+        settings->setBaseShadowTextureUnit(SceneUtil::MWShadow::baseShadowTextureUnit);
+        //settings->setMinimumShadowMapNearFarRatio(0);
+        settings->setNumShadowMapsPerLight(SceneUtil::MWShadow::numberOfShadowMapsPerLight);
+        //settings->setShadowMapProjectionHint(osgShadow::ShadowSettings::ORTHOGRAPHIC_SHADOW_MAP);
+        //settings->setMultipleShadowMapHint(osgShadow::ShadowSettings::PARALLEL_SPLIT); // ignored
+        //settings->setComputeNearFarModeOverride(osg::CullSettings::COMPUTE_NEAR_FAR_USING_PRIMITIVES);
+        //settings->setDebugDraw(true); // don't turn this on because it makes everything break
+
+        //settings->setPerspectiveShadowMapCutOffAngle(0);
+        //settings->setShaderHint(osgShadow::ShadowSettings::PROVIDE_VERTEX_AND_FRAGMENT_SHADER);
+
+        int mapres = 2048;
+        settings->setTextureSize(osg::Vec2s(mapres,mapres));
+
+        SceneUtil::MWShadow* tech = new SceneUtil::MWShadow();
+        shadowedScene->setShadowTechnique(tech);
+
+        /*tech->setMaxFarPlane(0);
+        tech->setTextureSize(osg::Vec2s(mapres, mapres));
+        tech->setShadowTextureCoordIndex(1);
+        tech->setShadowTextureUnit(1);
+        tech->setBaseTextureCoordIndex(0);
+        tech->setBaseTextureUnit(0);*/
+
+        //mRootNode->addChild(sceneRoot);
+        shadowedScene->addChild(sceneRoot);
+        mRootNode->addChild(shadowedScene);
+
+        Shader::ShaderManager::DefineMap shadowDefines = tech->getShadowDefines();
+        Shader::ShaderManager::DefineMap globalDefines = mResourceSystem->getSceneManager()->getShaderManager().getGlobalDefines();
+        globalDefines.insert(shadowDefines.begin(), shadowDefines.end());
```

Comment:
I would have liked this to be `insert_or_assign` to guarantee that things get replaced where necessary, but that requires C++ 17, which we don't use yet. I'm pretty sure we want the semantics of `insert_or_assign` here, so I guess I'll have to implement something that does the same thing.